tcping.go

```diff
@@ -238,31 +244,29 @@ func checkLatestVersion() {
 }
 
 /* Hostname resolution */
-func resolveHostname(tcpStats *stats) ipAddress {
-	ipRaw := net.ParseIP(tcpStats.hostname)
-
-	if ipRaw != nil {
-		return ipRaw.String()
-	}
+func resolveHostname(tcpStats *stats) (ipAddress, error) {
+    ip, err := netip.ParseAddr(tcpStats.hostname)
+    if err == nil {
+        return ip, nil
```

Comment:
since we are not really benefitting from the returned `err` in other functions, it is better to revert the error handling logic to the previous state.

if the resolve fails, we just handle it here. This way, we write less code in `processUserInput` function and also no need to put it in the trash in `retryResolve` function. Eventually, we minimize our program exit points.

Reply:
The reason I don't handle the error in function is that os.Exit prevents testing(there are ways to test it but I do not like them). To revert it back I need to remove the tests I have added to. A solution I think is replacing os.Exit with a panic but you can not pretty print using panic this time. We could revert back  if you are ok with no tests or discuss if you have another idea.

Reply:
I see. 
At the moment, nothing comes to mind to handle this in a better way.  Since it is for the most part a function that utilizes a standard library, maybe it does not hurt us too much to stick to a leaner and cleaner code base with less tests :S.